### PR TITLE
server: support autoid/leader/resign http API for AUTO_ID_CACHE=1

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: ref #52600

Problem Summary:

### What changed and how does it work?

Provide the http API to check and resign the auoid service leader.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [X] Manual test (add detailed scripts or steps below)

Not easy to write an integration test.
We need real tikv to start autoid service, and we need start tidb server to expose http status port.
There are no such utilities now, so I test it manually.

```
tiup playground  --mode tikv-slim
./bin/tidb-server  -store=tikv -path=127.0.0.1:2379
```

```
curl http://127.0.0.1:10080/autoid/leader
"192.168.0.16:10080"

curl http://127.0.0.1:10080/autoid/leader/resign
"success!"
```

- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by bb7133** [comment]
Why do you want to add this? For an easier debug?

**Comment by tiancaiamao** [comment]
> Why do you want to add this? For an easier debug?

Not only for debug, but also preparing for some scenario like the tidb leader node in wrong state, we can have some way to workaround.
Leader change is inevitable anyway.

@bb7133

File: pkg/server/handler/autoid_handler.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	}
+}
+
+// AutoIDLeaderHandler is the http handler for ${TIDB_HOST:PORT}//autoid/leader
+type AutoIDLeaderHandler func() owner.Manager
+
+// ServeHTTP implements http.Handler
+func (h AutoIDLeaderHandler) ServeHTTP(w http.ResponseWriter, _ *http.Request) {
+	om := h()
+	ddlOwnerID, err := om.GetOwnerID(context.Background())
```
Comment (zimulala) [inline pkg/server/handler/autoid_handler.go:48]: Do we need to check the `s.autoIDService` is nil in `h()`?

**Comment by tiancaiamao** [comment]
The resign owner API may be risky.
For example, this autoid service begin to do something because it's the owner, then, HTTP API called, make it losting its ownership, it break the assumption.

**Pull request closed without merging**